### ISSUE TITLE
inferred error sets have strippable identity (missing documentation?)

### ISSUE THREAD

**Issue by rohlem** [ziglang/zig#20862]
### Zig Version

0.14.0-dev.617+208baa37c

### Steps to Reproduce and Observed Behavior

Repro:
```zig
fn foo() !void {}
comptime {
    const A = @typeInfo(@TypeOf(foo())).ErrorUnion.error_set;
    const B = error{};
    @compileLog(A == B); //false
    @compileLog((A || error{}) == B); //true
    @compileLog(@Type(@typeInfo(A)) == B); //true
}
```

### Expected Behavior

I didn't know that inferred error sets have identity that makes them compare unequal to other all error sets - it's not currently mentioned in [their langref section](https://ziglang.org/documentation/master/#Inferred-Error-Sets).

When looking for previous issues on this I found that all error sets used to have identity, which changed with implementing https://github.com/ziglang/zig/issues/11022 .
[A comment there](https://github.com/ziglang/zig/issues/11022#issuecomment-1063401845) documents that inferred error sets have identity "to ease the implementation of the language specification. It's problematic to make type equality force error set resolution."
If that's still the current stance on the issue, I can accept that it just needs to be documented.

**Comment by mlugg** [comment]
Yep, status quo is intended, and this is just missing documentation.
